### PR TITLE
lightning: clone default filter for new configs

### LINKED ISSUE #38151: Lightning Config Becomes Abnormal After Loading a TOML with Source Table Filter
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Write a simple program: 
* Construct two TOML configs: one with table filter set, the other without table filters.
* Construct a Lightning config using the TOML data with table filter.
* Construct a Lightning config again using the TOML data without the table filter.

### 2. What did you expect to see? (Required)
Both generated Lightning config should be OK.  The second Lightning config should use the default table filter.

### 3. What did you see instead (Required)
In the second Lightning config, the table filter is not the default table filters.  It contains the filter item in the previous Lightning config.

### 4. What is your TiDB version? (Required)
Latest.

### PR DESCRIPTION
**Pull request by dsdashun**
### What problem does this PR solve?

Issue Number: close #38151 

Problem Summary:

### What is changed and how it works?
When constructing a new Lightning config, clone the default filter before assigning.  To avoid the elements of default filter being changed.  

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by dsdashun** [comment]
BR integration test passed: https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_integration_br_test/detail/tidb_ghpr_integration_br_test/10674/pipeline

File: br/pkg/lightning/config/config_test.go [hunk 1]
```diff
@@ -955,3 +956,41 @@ func TestCheckAndAdjustForLocalBackend(t *testing.T) {
 	cfg.TikvImporter.SortedKVDir = base
 	require.NoError(t, cfg.CheckAndAdjustForLocalBackend())
 }
+
+func TestCreateSeveralConfigsWithDifferentFilters(t *testing.T) {
```
Comment (okJiang) [inline br/pkg/lightning/config/config_test.go:960]: please add a false case

**Review by okJiang — COMMENTED** [review]
rest lgtm

File: br/pkg/lightning/config/config_test.go (hunk 1 above)
Comment (dsdashun) [reply br/pkg/lightning/config/config_test.go:960]: It has already included the false case.  If you change the implementation of `GetDefaultFilter()` by just returning `defaultFilter`, you'll find this test won't pass.

**Review by lance6716 — APPROVED** [review]

File: br/pkg/lightning/config/config.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ const (
 var (
 	supportedStorageTypes = []string{"file", "local", "s3", "noop", "gcs", "gs"}
 
-	DefaultFilter = []string{
+	defaultFilter = []string{
```
Comment (niubell) [inline br/pkg/lightning/config/config.go:101]: How about change this variable from `var` to `const`

File: br/pkg/lightning/config/config_test.go (hunk 1 above)
Comment (okJiang) [reply br/pkg/lightning/config/config_test.go:960]: yes, you are right. You can add what you said into this ut.

**Review by D3Hunter — APPROVED** [review]

File: br/pkg/lightning/config/config.go (hunk 2 above)
Comment (dsdashun) [reply br/pkg/lightning/config/config.go:101]: Golang doesn't support constant string slice.  Compile error:  `[]string{…} (value of type []string) is not constant`

**Pull request merged**